apps/myaccount/src/components/profile/profile.tsx

```diff
@@ -185,33 +184,91 @@ export const Profile: FunctionComponent<ProfileProps> = (props: ProfileProps): J
 
         const schemaNames = formName.split(".");
 
-        if (schemaNames.length === 1) {
-            if (isExtended) {
-                value = {
-                    [ProfileConstants.SCIM2_ENT_USER_SCHEMA]: {
-                        [schemaNames[0]]: values.get(formName)
-                    }
+        if (ProfileUtils.isMultiValuedSchemaAttribute(profileSchema, schemaNames[0]) ||
+            schemaNames[0] == "phoneNumbers") {
+            const attributeValues = [];
+
+            if (schemaNames.length === 1) {
+                // List of sub attributes.
+                const subValue = profileDetails.profileInfo[schemaNames[0]]
+                    && profileDetails.profileInfo[schemaNames[0]]
+                        .filter((subAttribute) => typeof subAttribute ==  "object");
+
+                if (subValue && subValue.length > 0) {
+                    subValue.map((value) => {
+                        attributeValues.push(value);
+                    });
                 }
-            } else {
-                value = schemaNames[0] === "emails"
-                    ? { emails: [values.get(formName)] }
-                    : { [schemaNames[0]]: values.get(formName) };
-            }
-        } else {
-            if (schemaNames[0] === "name") {
+
+                // This is required as the api doesn't support
+                // patching the attribute at the sub attribute level.
                 value = {
-                    name: { [schemaNames[1]]: values.get(formName) }
+                    [schemaNames[0]]: [
+                        ...attributeValues,
+                        values.get(formName)
+                    ]
                 };
             } else {
+                let primaryValue = "";
+
+                // The primary value of the email attribute.
+                if (schemaNames[0] == "emails") {
+                    primaryValue = profileDetails.profileInfo[schemaNames[0]]
```

Comment:
Better to check if `profileDetails.profileInfo` is defined before the operations to avoid errors.